arbos/block_processor.go

```diff
@@ -115,64 +115,101 @@ func createNewHeader(prevHeader *types.Header, l1info *L1Info, state *arbosState
+type SequencingHooks interface {
+	NextTxToSequence() (*types.Transaction, *arbitrum_types.ConditionalOptions, error)
+	DiscardInvalidTxsEarly() bool
+	SequencedTx(int) (*types.Transaction, error)
+	PreTxFilter(*params.ChainConfig, *types.Header, *state.StateDB, *arbosState.ArbosState, *types.Transaction, *arbitrum_types.ConditionalOptions, common.Address, *L1Info) error
+	PostTxFilter(*types.Header, *state.StateDB, *arbosState.ArbosState, *types.Transaction, common.Address, uint64, *core.ExecutionResult) error
+	BlockFilter(*types.Header, *state.StateDB, types.Transactions, types.Receipts) error
+	GetTxErrors() []error
+	InsertLastTxError(error)
+	ClearTxErrors()
 }
 
-type noopTxScheduler struct {
-	txs               types.Transactions
-	scheduledTxsCount int
+type NoopSequencingHooks struct {
+	txs                    types.Transactions
+	scheduledTxsCount      int
+	isBlockFilter          bool
+	isTxFilter             bool
+	discardInvalidTxsEarly bool
+	txErrors               []error
 }
 
-func (s *noopTxScheduler) GetNextTx() (*types.Transaction, error) {
+func (n *NoopSequencingHooks) NextTxToSequence() (*types.Transaction, *arbitrum_types.ConditionalOptions, error) {
 	// This is not supposed to happen, if so we have a bug
-	if s.scheduledTxsCount > len(s.txs) {
-		return nil, errors.New("noopTxScheduler: requested too many transactions")
+	if n.scheduledTxsCount > len(n.txs) {
+		return nil, nil, errors.New("noopTxScheduler: requested too many transactions")
 	}
-	if s.scheduledTxsCount == len(s.txs) {
-		return nil, nil
+	if n.scheduledTxsCount == len(n.txs) {
+		return nil, nil, nil
 	}
-	s.scheduledTxsCount += 1
-	return s.txs[s.scheduledTxsCount-1], nil
+	n.scheduledTxsCount += 1
+	return n.txs[n.scheduledTxsCount-1], nil, nil
 }
 
-func (s *noopTxScheduler) GetScheduledTx(txId int) (*types.Transaction, error) {
+func (n *NoopSequencingHooks) DiscardInvalidTxsEarly() bool {
+	return n.discardInvalidTxsEarly
+}
+
+func (n *NoopSequencingHooks) SequencedTx(txId int) (*types.Transaction, error) {
 	// This is not supposed to happen, if so we have a bug
-	if txId > len(s.txs) {
+	if txId > len(n.txs) {
 		return nil, errors.New("transaction queried for does not exist in the noopTxScheduler")
 	}
 	// This is not supposed to happen, if so we have a bug
-	if txId > s.scheduledTxsCount {
+	if txId > n.scheduledTxsCount {
 		return nil, errors.New("transaction queried for was not scheduled by the noopTxScheduler")
 	}
-	return s.txs[txId], nil
+	return n.txs[txId], nil
+}
+
+func (n *NoopSequencingHooks) PreTxFilter(config *params.ChainConfig, header *types.Header, db *state.StateDB, a *arbosState.ArbosState, transaction *types.Transaction, options *arbitrum_types.ConditionalOptions, address common.Address, info *L1Info) error {
+	if n.isTxFilter {
```

Comment:
this is part of a specific test and should not be part of the general code.
For NoopSequencingHooks this func should just return nil.